build/main_test.go

```diff
@@ -34,3 +34,45 @@ func ExampleTransactionBuilder() {
 	fmt.Printf("tx base64: %s", txeB64)
 	// Output: tx base64: AAAAADZY/nWY0gx6beMpf4S8Ur0qHsjA8fbFtBzBx1cbQzHwAAAAZAAAAAAAAAABAAAAAAAAAAAAAAABAAAAAAAAAAEAAAAALSRpLtCLv2eboZlEiHDSGR6Hb+zZL92fbSdNpObeE0EAAAAAAAAAAB3NZQAAAAAAAAAAARtDMfAAAABA2oIeQxoJl53RMRWFeLB865zcky39f2gf2PmUubCuJYccEePRSrTC8QQrMOgGwD8a6oe8dgltvezdDsmmXBPyBw==
 }
+
+// ExamplePathPayment creates and signs a simple transaction with PathPayment operation, and then
+// encodes it into a base64 string capable of being submitted to stellar-core.
+func ExamplePathPayment() {
```

Comment:
I agree with all of your points but I don't like the way we're going to define [send, dest] x [assets, amounts]. First, I think that with this API it's quite easy to mistake send and dest amounts ("was it `Source()` or `Dest()` function?"). Second, as you said, it looks a bit awkward.

What if `Path` mutator was a struct like this (similar thing to what @irisli did in Laboratory):

```go 
type Path struct {
	SendMax string
	DestinationAmount string
	Assets []{}interface // This will always have at least 2 elements: send and dest asset
}

type (p Path) setAssets(assets ...interface{}) Path {
   p.Assets = assets
   return p
}
```

Then, when it comes to creating a path we could do something like this:
```go
Payment(
	Destination{"GBDT3K42LOPSHNAEHEJ6AVPADIJ4MAR64QEKKW2LQPBSKLYD22KUEH4P"},
	Path{
		SendMax: "10",
		DestAmount: "20",
	}.setAssets(NativeAsset{}, CreditAsset{"USD", "Blah"}),
)
```
Passing `Path` mutator to a PaymentBuilder will switch to underlying `PathPaymentOp`.